src/main/java/net/fortuna/ical4j/model/component/Observance.java

```diff
@@ -155,12 +155,21 @@ public final Date getLatestOnset(final Date date) {
 
         if (initialOnset == null) {
             try {
-                initialOnset = applyOffsetFrom(calculateOnset(((DtStart) getProperty(Property.DTSTART)).getDate()));
+                DtStart dtStart = (DtStart) getProperty(Property.DTSTART);
```

Comment:
Using Optional would be the preferred approach given Java 8 minimum, however to continue to support Java 7 I'll introduce the getRequiredProperty method for now. Another option would be to use guava lib to support Optional, and perhaps will do that after the 2.0 release.